### PR TITLE
feat: Automatic documentation setup (Sphinx)

### PR DESCRIPTION
**Pull request by rpsimeon34**
This allows for semi-automatic documentation generation. We'll need to write docstrings in ising2D/src python files, following [reStructuredText](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html) formatting, but then it should build semi-automatically. HTML doc files go in ising2D/docs/build/html

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,6 +4,11 @@
 
 - There is lots to change here; this package is the inital commit as of 14 November 2023
 
-- To run the tests, make sure you have installed optional dependencies (via `pip install ising2D.["test"]`), and then run `pytest`
+- To run the tests, make sure you have installed test optional dependencies (via `pip install ising2D.["test"]`), and then run `pytest`
 
 - Before each commit, run `pre-commit run -a` to check the formatting.
+
+- To generate new html docuemtnation, do the following:
```
Comment (mirarenee) [inline README.md]: Typo in documentation

File: README.md [hunk 2]
```diff
@@ -4,6 +4,11 @@
...
+    - Make sure you installed docs optional dependencies (via `pip install ising2D.["docs"]`)
```
Comment (mirarenee) [inline README.md:12]: `pip install ising2D.["docs"]` (or rather `pip install .[docs]` as it works for me) doesn't run anything. Does this still need to be incorporated into the `pyproject.toml` file?

Comment (rpsimeon34) [reply README.md:12]: Thanks for catching that! Looks like I just forgot to include that in my commit. Should be fixed now

Comment (mirarenee) [reply README.md:12]: Looks good to merge!

**Pull request merged**